selfservice/flow/settings/error.go

```diff
@@ -163,9 +165,7 @@ func (s *ErrorHandler) WriteFlowError(
 		if shouldRespondWithJSON {
 			s.d.Writer().WriteError(w, r, aalErr)
 		} else {
-			http.Redirect(w, r, urlx.CopyWithQuery(
-				urlx.AppendPaths(s.d.Config().SelfPublicURL(r.Context()), login.RouteInitBrowserFlow),
-				url.Values{"aal": {string(identity.AuthenticatorAssuranceLevel2)}}).String(), http.StatusSeeOther)
+			http.Redirect(w, r, aalErr.RedirectTo, http.StatusSeeOther)
```

Comment:
Doesn't this significantly change the current behavior? Before, a user would be taken to the login screen to increase the AAL, now they just get redirected to some other page and the settings flow is not completed.

Reply:
Ah I see now, the error carries the `return_url`, which is the login URL, right? Is it always the login URL, or are there cases were it is different? Or do we just pipe through the original return URL?

Reply:
The `RedirectTo` URL is set by the `DoesSessionSatisfy` method which sets it to the login route and if applicable pipes the request_url into the `return_to` query parameter

Reply:
Please take a look at this part of the code https://github.com/ory/kratos/pull/3263/files#diff-a50a45cf2bc8e265082b551b7fb148ba52b47e6e6e1e0266d388bbd7955e20ccR335

that is where the "magic" happens